velox/common/hyperloglog/SparseHll.cpp

```diff
@@ -162,15 +162,20 @@ SparseHll::SparseHll(const char* serialized, HashStringAllocator* allocator)
 }
 
 void SparseHll::mergeWith(const SparseHll& other) {
-  mergeWith(other.entries_.size(), other.entries_.data());
+  auto size = other.entries_.size();
+  if (size) {
+    mergeWith(size, other.entries_.data());
+  }
 }
 
 void SparseHll::mergeWith(const char* serialized) {
   auto stream = initializeInputStream(serialized);
 
   auto size = stream.read<int16_t>();
-  mergeWith(
-      size, reinterpret_cast<const uint32_t*>(serialized + stream.offset()));
+  if (size) {
```

Comment:
Why we would merge with an empty sketch?  Sketch without data should be marked as null, no?

Reply:
I see, an empty sketch is created explicitly in the query.  Can we put some comments pointing out this is for  merging with explicit `empty_approx_set()` and should not happen in normal aggregation operator?